vite.config.js

```diff
@@ -0,0 +1,24 @@
+import { defineConfig } from "vite";
+import react from "@vitejs/plugin-react";
+import { NodeGlobalsPolyfillPlugin } from "@esbuild-plugins/node-globals-polyfill";
+import { NodeModulesPolyfillPlugin } from "@esbuild-plugins/node-modules-polyfill";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
+  optimizeDeps: {
+    esbuildOptions: {
+      // Node.js global to browser globalThis
+      define: {
+        global: "globalThis",
+      },
+      // Enable esbuild polyfill plugins
+      plugins: [
+        NodeGlobalsPolyfillPlugin({
+          buffer: true,
+        }),
+        NodeModulesPolyfillPlugin(),
+      ],
```

Comment:
Interesting, we'd been using `vite-plugin-node-polyfills` on trefoil-web vite migration...
Didn't know about these officially supported plugins... maybe an advantage to those.

Also, I think its fine to add the plugins here but it just gets flattened into top level plugins array (the one you have with `[react()]`) anyways.

Reply:
Yeah I tried moving them up into that plugins array and everything broke again. Not sure why it seemingly needs to be `esbuildOptions`. I'll make a note to look into that and see if I can get rid of those extra objects and put everything top level.